### ISSUE TITLE
如何debug训练过程

### ISSUE THREAD

**Issue by YYGe01** [universal-ie/UIE#16]
作者您好，感谢开源,  我正在运行您的代码，有两个问题.
1.  我在linux环境下运行代码,发现这里的所有bash脚本运行都会报错:, 如:

```
>  run_seq2seq_record.bash: line 8: $'\r': command not found
> : No such file or directoryne 9: scripts/function_code.bash
> run_seq2seq_record.bash: line 10: $'\r': command not found
> run_seq2seq_record.bash: line 11: syntax error near unexpected token `$'do\r''
> 'un_seq2seq_record.bash: line 11: `for index in $(seq 1 ${run_time}); do
```
看网上说是不同的平台\r\n格式的问题, 已经粗暴的替换格式解决.

2. 训练如何进行单步调试呢

```
bash run_seq2seq_record.bash -v -d 0 \
  -b 16 \
  -k 3 \
  --lr 1e-4 \
  --warmup_ratio 0.06 \
  -i absa/14lap \
  --epoch 50 \
  --spot_noise 0.1 \
  --asoc_noise 0.1 \
  -f spotasoc \
  --epoch 50 \
  --map_config config/offset_map/closest_offset_en.yaml \
  -m hf_models/uie-base-en \
  --random_prompt
```
水平有限, 问下这个bash脚本如何可以进入 run_seq2seq.py 中进行断点单步调试呢, 谢谢!

**Comment by luyaojie** [comment]
您好，谢谢反馈。

1. 请问您的环境是什么，我这边很多环境运行确实没遇到这个问题，我看如何统一一下
2. 我也没在命令行中单步调试过，不过我了解 pdb 应该可以实现单步调试的需求

**Comment by YYGe01** [comment]
好的,谢谢!

**Comment by SnowBoy0215** [comment]
请问，您现在可以逐步debug微调过程了吗

**Comment by YYGe01** [comment]
> 

可以了, 用的类似您推荐的第二种方法, 具体为, 在run_seq2seq_record.bash脚本中添加如下:
`
  echo " 
    --do_train --do_eval --do_predict ${constraint_decoding} ${fp16} \
    --use_fast_tokenizer=True \
    --ddp_find_unused_parameters=False \
    --predict_with_generate \
    --evaluation_strategy=${evaluation_strategy} \
    --save_strategy=${evaluation_strategy} \
    --metric_for_best_model eval_overall-F1 \
    --save_total_limit 1 \
    --load_best_model_at_end \
    --max_source_length=${max_source_length:-"256"} \
    --max_prefix_length=${max_prefix_length:-"-1"} \
    --max_target_length=${max_target_length:-"192"} \
    --num_train_epochs=${epoch} \
    --task=${task_name} \
    --train_file=${data_folder}/train.json \
    --validation_file=${data_folder}/val.json \
    --test_file=${data_folder}/test.json \
    --record_schema=${data_folder}/record.schema \
    --per_device_train_batch_size=${batch_size} \
    --per_device_eval_batch_size=$((batch_size * 4)) \
    --output_dir=${output_dir} \
    --logging_dir=${output_dir}_log \
    --model_name_or_path=${model_name} \
    --learning_rate=${lr} \
    --source_prefix="${task_name}: " \
    --lr_scheduler_type=${lr_scheduler} \
    --label_smoothing_factor=${label_smoothing} \
    --eval_steps ${eval_steps} \
    --decoding_format ${decoding_format} \
    --warmup_ratio ${warmup_ratio} \
    --preprocessing_num_workers=4 \
    --dataloader_num_workers=0 \
    --meta_negative=${negative} \
    --meta_positive_rate=${positive} \
    --skip_memory_metrics \
    --no_remove_unused_columns \
    --ordered_prompt=${ordered_prompt} \
    --save_better_checkpoint=False \
    --start_eval_step=${start_eval_step:-"0"} \
    --spot_noise=${spot_noise} \
    --asoc_noise=${asoc_noise} \
    --seed=${seed}${index} --disable_tqdm=${disable_tqdm}"

就可以打印run_seq2seq.py 的参数了,可以进行逐步debug了.

**Comment by SnowBoy0215** [comment]
您是说只加了echo “”，可以在py文件中debug吗？是否可以留个邮箱向您咨询一下。
